### PR TITLE
[Spring] Spring generator uses old Swagger dependency

### PR DESCRIPTION
**Pull request by alexsuperdev**
Fix [4245](https://github.com/OpenAPITools/openapi-generator/issues/4245)

Updated swagger-annotations dependency from version 1.5.23 to the actual last release 2.1.1
in generated code for spring libraries
So from

```java
<dependency>
    <groupId>io.swagger</groupId>
    <artifactId>swagger-annotations</artifactId>
    <version>1.5.23</version>
</dependency>
```
To:
```java
<dependency>
    <groupId>io.swagger.core.v3</groupId>
    <artifactId>swagger-annotations</artifactId>
    <version>2.1.1</version>
</dependency>
```

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

cc @bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10) @bkabrda (2020/01)

### CODE REVIEW

**Comment by alexsuperdev** [comment]
Is there a chance that this pullrequest will be reviewed already in this month?

**Comment by alexsuperdev** [comment]
@wing328 Can you please review this peace of code?

**Comment by shred22** [comment]
Any tentative timeline on this PR merge ?

**Comment by atkawa7** [comment]
@wing328 ping

**Comment by piotrkav** [comment]
any update on this?

**Comment by cbornet** [comment]
The problem is I don't think Sprinfox v2 which is a widely used lib for Spring support OASv3 annotations.
Sprinfox v3 has been released but I don't think the adoption is significant yet.

**Comment by piotrkav** [comment]
> The problem is I don't think Sprinfox v2 which is a widely used lib for Spring support OASv3 annotations.
> Sprinfox v3 has been released but I don't think the adoption is significant yet.

Sorry for misunderstanding, but how springfox is related in this matter? 
Using the openapi-generator-maven-plugin it would be ideal if it would generate api interfaces using 
```java
<dependency>
    <groupId>io.swagger.core.v3</groupId>
    <artifactId>swagger-annotations</artifactId>
    <version>2.1.1</version>
</dependency>
```
which this PR fixes.
According to https://openapi-generator.tech/docs/plugins#dependencies there is a possibility to use v3 annotations, but it doesn't work.

**Comment by cbornet** [comment]
These annotations are used by libraries to generate the OAS. Springfox is I think the most widely used lib in the Spring world. Or do you have other libs in mind ?

**Comment by piotrkav** [comment]
> These annotations are used by libraries to generate the OAS. Springfox is I think the most widely used lib in the Spring world. Or do you have other libs in mind ?

Springfox is not required to use openapi-generator-maven-plugin. That's why it confuses me 😄  

 I was thinking about using the springdoc (which would require the v3 annotations to actually show examples/description) and redoc for static files.

**Comment by cbornet** [comment]
This PR is not about `openapi-generator-maven-plugin`. It's about the generated code.
Yes Springdoc and Springfox v3 will work with the new annotations. But I think the large majority still uses Springfox v2.
Note that Springfox v3 works with the old annotations to generate an OASv3 doc.

**Comment by alexsuperdev** [comment]
> 
> 
> This PR is not about `openapi-generator-maven-plugin`. It's about the generated code.
> Yes Springdoc and Springfox v3 will work with the new annotations. But I think the large majority still uses Springfox v2.
> Note that Springfox v3 works with the old annotations to generate an OASv3 doc.


> 
> 
> This PR is not about `openapi-generator-maven-plugin`. It's about the generated code.
> Yes Springdoc and Springfox v3 will work with the new annotations. But I think the large majority still uses Springfox v2.
> Note that Springfox v3 works with the old annotations to generate an OASv3 doc.

Hello @cbornet . I tested it locally, it works with Springfox v2.

**Comment by jimschubert** [comment]
@cbornet this is specific to the JavaSpring generated output. Since we have a major release coming, I think it would make sense to update to springfox2 3.0.0 in JavaSpring and have this PR integrated in the 5.0 release. If that could be done here, it would also solve issues with pulling openapi-generator into a Spring project generated with our tooling (which [currently does not work](https://openapi-generator.slack.com/archives/CLSB0U0R5/p1596192414211500)). See #7102 which is related.

**Comment by cbornet** [comment]
> Hello @cbornet . I tested it locally, it works with Springfox v2.

Oh, great ! I thought this was only part of the v3. Then I have no more concern for this PR.

File: modules/openapi-generator/src/main/resources/JavaSpring/libraries/spring-boot/pom.mustache [hunk 1]
```diff
@@ -94,6 +95,11 @@
             <artifactId>springfox-swagger-ui</artifactId>
             <version>${springfox-version}</version>
         </dependency>
+        <dependency>
+          <groupId>io.swagger.core.v3</groupId>
```
Comment (cbornet) [inline modules/openapi-generator/src/main/resources/JavaSpring/libraries/spring-boot/pom.mustache:99]: The pom is aligned with 4 spaces

File: modules/openapi-generator/src/main/resources/JavaSpring/libraries/spring-mvc/pom.mustache [hunk 2]
```diff
@@ -165,6 +165,11 @@
             <artifactId>springfox-swagger-ui</artifactId>
             <version>${springfox-version}</version>
         </dependency>
+        <dependency>
+          <groupId>io.swagger.core.v3</groupId>
```
Comment (cbornet) [inline modules/openapi-generator/src/main/resources/JavaSpring/libraries/spring-mvc/pom.mustache:169]: alignment

File: modules/openapi-generator/src/main/resources/JavaSpring/libraries/spring-mvc/pom.mustache [hunk 3]
```diff
@@ -173,9 +178,9 @@
             <version>3.14.2</version>
         </dependency>
         <dependency>
-            <groupId>io.swagger</groupId>
+          <groupId>io.swagger.core.v3</groupId>
```
Comment (cbornet) [inline modules/openapi-generator/src/main/resources/JavaSpring/libraries/spring-mvc/pom.mustache:181]: alignment

File: samples/client/petstore/spring-cloud-async/src/main/java/org/openapitools/api/PetApi.java [hunk 4]
```diff
@@ -41,18 +47,18 @@
...
-    @ApiOperation(value = "Add a new pet to the store", nickname = "addPet", notes = "", authorizations = {
-        @Authorization(value = "petstore_auth", scopes = {
-            @AuthorizationScope(scope = "write:pets", description = "modify pets in your account"),
-            @AuthorizationScope(scope = "read:pets", description = "read your pets")
-            })
-    }, tags={ "pet", })
-    @ApiResponses(value = { 
-        @ApiResponse(code = 405, message = "Invalid input") })
+    @Operation(summary = "Add a new pet to the store", operationId = "addPet", description = "",
+     tags={ "pet", },
```
Comment (cbornet) [inline samples/client/petstore/spring-cloud-async/src/main/java/org/openapitools/api/PetApi.java:51]: alignments (should be 4 spaces)

Comment (alexsuperdev) [reply samples/client/petstore/spring-cloud-async/src/main/java/org/openapitools/api/PetApi.java:51]: fixed

**Comment by atkawa7** [comment]
ping @wing328

**Comment by cbornet** [comment]
LGTM. @alexsuperdev can you resolve the conflicts ?

**Comment by alexsuperdev** [comment]
> 
> 
> LGTM. @alexsuperdev can you resolve the conflicts ?

I tried to resolve them, but not succeeded, its to much work to do, because to many changes were made in code since march. 
So in short, i will not resolve this issue. Sorry

**Comment by ddeath** [comment]
@wing328 I guess this one can be closed

**Pull request closed without merging**